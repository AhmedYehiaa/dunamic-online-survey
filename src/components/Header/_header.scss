.header {
  width: 100%;
  background: $off-black;
  color: white;
  margin-bottom: $xl-size;
  padding: $m-size $l-size;
}

.header__title {
  font-size: $l-size;
  margin: 0;
}

.header__subtitle {
  color: $off-white;
  font-size: $m-size;
  font-weight: 500;
  margin: 0;
}
