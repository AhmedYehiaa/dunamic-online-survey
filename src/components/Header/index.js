import React from 'react';
import PropTypes from 'prop-types';

const Header = (props) => {
  const { title, subTitle } = props;
  return (
    <header className="header">
      <h1 className="header__title">{title}</h1>
      <h2 className="header__subtitle">{subTitle}</h2>
    </header>
  );
};

Header.defaultProps = {
  title: '',
  subTitle: '',
};

Header.propTypes = {
  title: PropTypes.string,
  subTitle: PropTypes.string,
};

export default Header;
