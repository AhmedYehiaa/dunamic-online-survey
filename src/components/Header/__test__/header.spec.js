import React from 'react';
import { shallow } from 'enzyme';
import Header from '../index';

it('it should renders the radio options', () => {
  const wrapper = shallow(
    <Header
      title="title"
      subTitle="subtitle"
    />,
  );
  expect(wrapper.children.length).toEqual(1);
  expect(wrapper).toMatchSnapshot();
});
