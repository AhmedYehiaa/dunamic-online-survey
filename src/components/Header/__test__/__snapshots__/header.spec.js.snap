// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should renders the radio options 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    subTitle="subtitle"
    title="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="header__title"
        >
          title
        </h1>,
        <h2
          className="header__subtitle"
        >
          subtitle
        </h2>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "title",
          "className": "header__title",
        },
        "ref": null,
        "rendered": "title",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "subtitle",
          "className": "header__subtitle",
        },
        "ref": null,
        "rendered": "subtitle",
        "type": "h2",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="header__title"
          >
            title
          </h1>,
          <h2
            className="header__subtitle"
          >
            subtitle
          </h2>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "title",
            "className": "header__title",
          },
          "ref": null,
          "rendered": "title",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "subtitle",
            "className": "header__subtitle",
          },
          "ref": null,
          "rendered": "subtitle",
          "type": "h2",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
